StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028NoTrailingWhitespace.cs

```diff
@@ -0,0 +1,160 @@
+﻿namespace StyleCop.Analyzers.SpacingRules
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Text;
+
+    /// <summary>
+    /// Discovers any C# lines of code with trailing whitespace.
+    /// </summary>
+    /// <remarks>
+    /// <para>A violation of this rule occurs whenever the code contains whitespace at the end of the line.</para>
+    ///
+    /// <para>Trailing whitespace causes unnecessary diffs in source control,
+    /// looks tacky in editors that show invisible whitespace as visible characters,
+    /// and is highlighted as an error in some configurations of git.</para>
+    ///
+    /// <para>For these reasons, trailing whitespace should be avoided.</para>
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class SA1028NoTrailingWhitespace : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1028NoTrailingWhitespace"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "SA1028";
+        private const string Title = "Code must not contain trailing whitespace";
+        private const string MessageFormat = "Remove trailing whitespace";
+        private const string Category = "Style";
+        private const string Description = "There should not be any whitespace at the end of a line of code.";
+        private const string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/wiki/SA1028";
+        private const bool IsEnabledByDefault = true;
```

Comment:
:bulb: It would make sense (as a separate pull request) to create a new property `AnalyzerConstants.EnabledByDefault` for this, and use it consistently through the project. I created issue #783 for this.